crates/worker/src/partition_processor_manager.rs

```diff
@@ -645,35 +744,48 @@ impl<T: TransportConnect> PartitionProcessorManager<T> {
             CreateSnapshot(partition_id, sender) => {
                 self.running_partition_processors
                     .get(&partition_id)
-                    .map(|store| store.handle.create_snapshot(Some(sender)));
+                    .map(|store| {
+                        let ProcessorStatus::Started(state) = store else {
+                            return None;
+                        };
+
+                        Some(state.handle.create_snapshot(Some(sender)))
+                    });
             }
             GetState(sender) => {
                 let _ = sender.send(self.get_state());
             }
         }
     }
 
-    async fn on_control_processors(
+    #[instrument(parent=None, skip_all)]
+    fn on_control_processors(
         &mut self,
-        control_processor: Incoming<ControlProcessors>,
-    ) -> Result<(), Error> {
-        let control_processors = control_processor.into_body();
-
-        self.metadata
-            .wait_for_version(
-                MetadataKind::Logs,
-                control_processors.min_logs_table_version,
-            )
-            .await?;
-        let partition_table = self
-            .metadata
-            .wait_for_partition_table(control_processors.min_partition_table_version)
-            .await?
-            .into_arc();
-
-        for control_processor in control_processors.commands {
-            self.on_control_processor(control_processor, &partition_table)
-                .await?;
+        control_processors: ControlProcessors,
+        effects: mpsc::UnboundedSender<ManagerEffect>,
+    ) -> Result<(), ShutdownError> {
+        self.control_processors = None;
```

Comment:
nit: Instead of passing in the `control_processors` message here and then clearing `self.control_processors` as well as taking when an update of the version happens and then reassign the value if it does not fulfill the versions, we could assign to `self.control_processors` when we receive a new `control_processors` in the `self.incoming_update_processors.next()` arm and then call `self.on_control_processors()`. That way the invariant would be if `self.control_processors` is `Some`, then do the check and clear the field if the command could be applied.

Reply:
You are right! Will change it.